lib/query-types.d.ts

```diff
@@ -1,16 +1,17 @@
+declare enum QueryTypes {
```

Comment:
it should be exported, not declared as a global

Reply:
I tried that by using `export default enum QueryTypes { ....` but the compiler complains with the message `Expression expected. [1109]`.

It doesn't seem like I can use `const enum QueryTypes {..` followed by `export default QueryTypes` because of the error: `A 'declare' modifier is required for a top level declaration in a .d.ts file. [1046]`.

Any suggestions?

Reply:
Ah, I see. I think I can use `export enum QueryTypes {...` here. I'll push up a fix....

Reply:
Please make sure to mirror the export style in the Sequelize library.

Reply:
Are you referring to me using `export default ...` instead of `export ...`? If so, that makes sense and I'll push up a fix soon.

If you are referring to something else in the library, then let me know if I'm missing anything.

Reply:
The QueryTypes are exported with `export =`. Please maintain that.

Reply:
@felixfbecker I'm not sure that I can export my enum using the `export =` assignment without declaring my enum globally. When I try the following:

```
enum QueryTypes {
  SELECT = "SELECT",
  INSERT = "INSERT",
  UPDATE = "UPDATE",
  BULKUPDATE = "BULKUPDATE",
  BULKDELETE = "BULKDELETE",
  DELETE = "DELETE",
  UPSERT = "UPSERT",
  VERSION = "VERSION",
  SHOWTABLES = "SHOWTABLES",
  SHOWINDEXES = "SHOWINDEXES",
  DESCRIBE = "DESCRIBE",
  RAW = "RAW",
  FOREIGNKEYS = "FOREIGNKEYS",
}

module.exports = QueryTypes
```

I get the following error: `A 'declare' modifier is required for a top level declaration in a .d.ts file. [1046]`.

And when I try:

```
module.exports = enum {
  SELECT = "SELECT",
  INSERT = "INSERT",
  UPDATE = "UPDATE",
  BULKUPDATE = "BULKUPDATE",
  BULKDELETE = "BULKDELETE",
  DELETE = "DELETE",
  UPSERT = "UPSERT",
  VERSION = "VERSION",
  SHOWTABLES = "SHOWTABLES",
  SHOWINDEXES = "SHOWINDEXES",
  DESCRIBE = "DESCRIBE",
  RAW = "RAW",
  FOREIGNKEYS = "FOREIGNKEYS",
}
```

I get this error: `Expression expected. [1109]`.

Since enums are types and cannot be assigned within an expression, this seems to be the same reason why we are using `export interface ...` instead of `exports = interface ...`.

What do you think?

Reply:
Did you try
```ts
enum QueryTypes { ... }
export = QueryTypes
```

Reply:
Yes, and I get the same `A 'declare' modifier is required for a top level declaration in a .d.ts file. [1046]` error